pkg/catalog/catalog.go

```diff
@@ -891,22 +893,70 @@ func (c *Catalog) ListCommits(ctx context.Context, repository string, branch str
 		for _, parent := range v.Parents {
 			commit.Parents = append(commit.Parents, parent.String())
 		}
-		commits = append(commits, commit)
-		if len(commits) >= limit+1 {
+
+		if len(params.PathList) != 0 && len(v.Parents) == 1 {
```

Comment:
You may want to use one of the following suggestions to increase readability

```suggestion
		if len(params.PathList) != 0 && len(v.Parents) == NumberOfParentsOfNonMergeCommit {
```

```suggestion
		if len(params.PathList) != 0 && !isMergeCommit(v) {
```

Reply:
And also, can you explicitly check if the objects or prefix query params are there instead of checking the length of params list?